clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRToMLIR.cpp

```diff
@@ -709,6 +709,145 @@ class CIRGetGlobalOpLowering
   }
 };
 
+class CIRCastOpLowering : public mlir::OpConversionPattern<mlir::cir::CastOp> {
+public:
+  using OpConversionPattern<mlir::cir::CastOp>::OpConversionPattern;
+
+  inline mlir::Type convertTy(mlir::Type ty) const {
+    return getTypeConverter()->convertType(ty);
+  }
+
+  /// If the given type is a vector type, return the vector's element type.
+  /// Otherwise return the given type unchanged.
+  inline mlir::Type elementTypeIfVector(mlir::Type type) const {
```

Comment:
If you are adding logic that include handling vectors, you should add some tests for them too.

Reply:
It seems cir.alloca for vector and passing as arguments are not fully supported yet. Perhaps introduce the logic after we could test them well?

Reply:
Sounds good too, please make sure you emit an error if vectors are used.

Reply:
Done. Thanks.